SDL2pp/Surface.cc

```diff
@@ -41,16 +41,33 @@ Surface::Surface(SDL_Surface* surface) : surface_(surface) {
 	assert(surface);
 }
 
-Surface::Surface(Uint32 flags, int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
-	if ((surface_ = SDL_CreateRGBSurface(flags, width, height, depth, Rmask, Gmask, Bmask, Amask)) == nullptr)
+Surface::Surface(int width, int height, int depth, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
+	// Flags are obsolete and should be set to 0 according to the documentation
+	if ((surface_ = SDL_CreateRGBSurface(0, width, height, depth, Rmask, Gmask, Bmask, Amask)) == nullptr) {
 		throw Exception("SDL_CreateRGBSurface");
+	}
 }
 
 Surface::Surface(void* pixels, int width, int height, int depth, int pitch, Uint32 Rmask, Uint32 Gmask, Uint32 Bmask, Uint32 Amask) {
 	if ((surface_ = SDL_CreateRGBSurfaceFrom(pixels, width, height, depth, pitch, Rmask, Gmask, Bmask, Amask)) == nullptr)
 		throw Exception("SDL_CreateRGBSurfaceFrom");
 }
 
+#if SDL_VERSION_ATLEAST(2, 0, 5)
+Surface::Surface(int width, int height, int depth, Uint32 format) {
+	// Flags are obsolete and should be set to 0 according to the documentation
+	if ((surface_ = SDL_CreateRGBSurfaceWithFormat(0, width, height, depth, format)) == nullptr) {
```

Comment:
I general I just agree with your style despite some of it is not what I like/think is better in the base that most of it would be just opinionated discussions, but here I will push back.

After the [#gotofail](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/) issue from some years back I consider is a good practice to put braces around the code you are making conditional, even if is just one line.

I know this won't cause a security hole that will impact such a big range of users, but still, is good to say when the if should end, is just two extra characters one extra line